src/network/networkd-link.c

```diff
@@ -689,7 +689,9 @@ static Address* link_find_dhcp_server_address(Link *link) {
 static int link_enter_configured(Link *link) {
         assert(link);
         assert(link->network);
-        assert(link->state == LINK_STATE_SETTING_ROUTES);
+
+        if (link->state != LINK_STATE_SETTING_ROUTES)
+                return 1;
```

Comment:
Thanks ! But we already using this same syntax other places

```
link_down_handler
    if (IN_SET(link->state, LINK_STATE_FAILED, LINK_STATE_LINGER))
                return 1;
```

Should we all change that too.